### PR TITLE
Improve the “Contribute Code” docs

### LINKED ISSUE #2463: Document how everything is documented across the different modules
As a developer, I’d like an overview of how the docs are written in the individual repos and are generated into their READMEs & the [CanJS 3 docs](http://canjs.github.io/canjs/) so that I can more easily contribute documentation changes. In addition to just making changes, it’d be helpful to generate the docs and preview the changes locally.

Furthermore, if I’m creating a new project that would be a good fit for the [CanJS ecosystem page](http://canjs.github.io/canjs/doc/can-ecosystem.html), I’d like to know how I should document my project and propose that my project be included on that page.

### PR DESCRIPTION
**Pull request by chasenlehara**
- Fix a bunch of spelling, grammar, and punctuation issues
- Remove the `> ` in the command-line examples so they’re easier to copy + paste
- Organize the file structure alphabetically
- Add a note about manually upgrading the code from the DoneJS plugin generator to CanJS 3

Closes #2463

### CODE REVIEW

File: docs/can-guides/contribute/code.md [hunk 1]
```diff
@@ -7,59 +7,59 @@
...
+If you already have `git` installed, make sure you’ve
 [setup your ssh keys](https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/).
 
-### Get NodeJS
+### Get Node.js
 
-Download Node.js version 5 or later at [https://nodejs.org](https://nodejs.org).  You can
-verify node is working at its version with:
+Download Node.js version 5 or later at [NodeJS.org](https://nodejs.org).  You can
+verify Node’s version with:
```
Comment (justinbmeyer) [inline docs/can-guides/contribute/code.md]: the real goal is to verify node is working by printing out its version.

File: docs/can-guides/contribute/code.md [hunk 2]
```diff
@@ -68,151 +68,152 @@ Download the Firefox browser
...
-├── .npmignore              - Tells npm publish to ignore certain files
-├── .travis.yml             - Travis CI configuration
-├── readme.md               - Automatically generated readme
-├── test/                   - Test files
-|   ├── can-example-test.js - Main test file
-|   ├── test.html           - Main test page
-├── docs/                   - Documentation source
-|   ├── can-example.md      - Package or module documentation
-├── node_modules/           - Node dependency installation folder
+├── .editorconfig           — Configures editors for this project
```
Comment (justinbmeyer) [inline docs/can-guides/contribute/code.md]: The left was organized by importance.

File: docs/can-guides/contribute/code.md [hunk 3]
```diff
@@ -68,151 +68,152 @@ Download the Firefox browser
...
 1. Create a new feature branch. For example, `git checkout -b html5-fix`.
 2. Make some changes to the code and tests.
-4. Run tests `npm test` and make sure they pass in all browsers.
-5. Update documentation if necessary.
+4. Run `npm test` to make sure the tests pass in all browsers.
+5. Update the documentation if necessary.
 6. Push your changes to your remote branch.  For example, `git push origin html5-fix`.
-7. Submit a pull request! Navigate to Pull Requests and click the 'New Pull Request' button. Fill in some
-   details about your potential patch including a meaningful title. When finished, press "Send pull request". The core team will be notified about your submission and let you know of any problems or targeted release date.
+7. Submit a pull request! On GitHub, navigate to Pull Requests and click the “New Pull Request” button. Fill in some
```
Comment (justinbmeyer) [inline docs/can-guides/contribute/code.md]: I'm wary of using special quotes other than `'` or `"`.  `'` and `"` are easier to type and, at least for me, provide equivalently readable punctuation.  I don't want to be inconsistent.  I've noticed these changes mistakenly in code blocks, which breaks the code.

Comment (chasenlehara) [inline docs/can-guides/contribute/code.md]: Hey @justinbmeyer, a few comments on using the correct punctuation marks:
- It’s [pretty easy](http://www.danandcheryl.com/2010/08/how-to-type-curly-quotes-in-mac-os-x) to type the correct marks (although I don’t expect anyone else to care enough to change their habits).
- The correct marks are easier to read and can help make text more understandable.
- I’ve definitely seen issues with software automatically changing “dumb” quotes to “smart” quotes (ugh presentation software), so I wouldn’t really recommend automated replacement. Using the correct marks can actually fix some issues though, like in this PR:
before
after

Again, I’m not really expecting anyone else to care, but as I go through the docs, I’m going to clean up our spelling, grammar, typography, etc. It doesn’t take much time to get the little details right. 😊

**Pull request merged**